File: src/main/java/org/noxet/noxetserver/menus/book/BookMenu.java
package org.noxet.noxetserver.menus.book;

import net.md_5.bungee.api.chat.BaseComponent;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.BookMeta;

import java.util.List;

public class BookMenu {
    private final ItemStack book;

    public BookMenu(List<BaseComponent[]> pages) {
        book = new ItemStack(Material.WRITTEN_BOOK);
        BookMeta bookMeta = (BookMeta) book.getItemMeta();

        assert bookMeta != null;
        bookMeta.spigot().setPages(pages);

        bookMeta.setAuthor("NOXET");
        bookMeta.setTitle("Book Menu");

        book.setItemMeta(bookMeta);
    }

    public void openMenu(Player player) {
        player.openBook(book);
    }
}

File: src/main/java/org/noxet/noxetserver/menus/book/BookMenu.scala
package org.noxet.noxetserver.menus.book

import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.BookMeta

class BookMenu(pages: java.util.List[Array[BaseComponent]]):
  val book = ItemStack(Material.WRITTEN_BOOK)
  book.setItemMeta({
    val meta = book.getItemMeta.asInstanceOf[BookMeta]
    meta.spigot.setPages(pages)
    meta.setAuthor("NOXET")
    meta.setTitle("Book Menu")
    meta
  })

  def openMenu(player: Player): Unit = player.openBook(book)